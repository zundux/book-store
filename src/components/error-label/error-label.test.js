import React from "react";
import ErrorLabel from "./error-label";

describe("Error label", () => {

  test("Error label has only the message", () => {
    const message = "Something went wrong";

    const component = shallow(
      <ErrorLabel {...{
        message
      }}/>
    );
    matchSnapshot(component);
  });

  test("Error label has the message and button with text \"Retry\"", () => {
    const message = "Something went wrong";
    const onRetry = jest.fn();

    const component = shallow(
      <ErrorLabel {...{
        message,
        onRetry
      }}/>
    );
    matchSnapshot(component);
    // call func onRetry if click on Retry button
    component.find(".retry").simulate("click");
    expect(onRetry).toHaveBeenCalled();
  });
});

