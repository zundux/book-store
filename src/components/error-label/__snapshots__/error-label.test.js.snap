// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error label Error label has only the message 1`] = `
<div>
  <p>
    Something went wrong
  </p>
</div>
`;

exports[`Error label Error label has the message and button with text "Retry" 1`] = `
<div>
  <p>
    Something went wrong
  </p>
  <button
    className="retry"
    onClick={[Function]}
  >
    Retry
  </button>
</div>
`;
